webapp/channels/src/actions/file_actions.ts

```diff
@@ -88,10 +88,15 @@ export function uploadFile({file, name, type, rootId, channelId, clientId, onPro
 
                     onSuccess(response, channelId, rootId);
                 } else if (xhr.status >= 400 && xhr.readyState === 4) {
-                    const errorResponse = JSON.parse(xhr.response);
-                    const errorMessage =
+                    let errorMessage = '';
+                    try {
+                        const errorResponse = JSON.parse(xhr.response);
+                        errorMessage =
```

Comment:
This is more of a general question: Why are there cases where the sever returns a non-JSON response? Would it make the webapp code easier and more robust if the server would return a JSON response for xhr requests?

Reply:
I'm not exactly sure @hanzei, I could not reproduce this locally and it only seemed to happen on the spinwicks